File: backend/schemas/file.py
import uuid
from datetime import datetime
from pydantic import BaseModel


class UploadedFileOut(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    filename: str
    language: str
    size: int
    path: str
    created_at: datetime

    # Keep backward compatibility for existing frontend calls
    file_type: str
    size_bytes: int

    class Config:
        from_attributes = True


class UploadedFileWithContent(UploadedFileOut):
    content: str


class AnalyzeRequest(BaseModel):
    action: str

File: frontend/src/types/file.ts
export interface UploadedFile {
  id: string;
  user_id: string;
  filename: string;
  language: string;
  size: number;
  path: string;
  created_at: string;

  // Keep backward compatibility
  file_type: string;
  size_bytes: number;
}

export interface UploadedFileWithContent extends UploadedFile {
  content: string;
}
